org.eclipse.wildwebdeveloper/schema/org.eclipse.wildwebdeveloper.jsonschemaassociations.exsd

```diff
@@ -0,0 +1,238 @@
+This might be necessary when assigning a JSON schema to more than one file pattern or vice versa (latter might be less often the case).
+&lt;/p&gt;
+
+&lt;p&gt;
+&lt;strong&gt;Important hint for developing plug-ins using this extension point:&lt;/strong&gt;&lt;/br&gt;
+It is possible to use a JSON schema file, that is stored in another plug-in by declaring an additional bundle-id in the schema association.
+At the moment, it is necessary to unpack the bundle after installation to access the file at runtime.
+To achieve this, set the header &lt;code&gt;Eclipse-BundleShape&lt;/code&gt; to the value &lt;code&gt;dir&lt;/code&gt;.
+&lt;/br&gt;
+Requirements (MANIFEST.MF):
+&lt;ul&gt;
+	&lt;li&gt;&lt;code&gt;Require-Bundle: org.eclipse.wildwebdeveloper;bundle-version=&quot;0.3.0&quot;&lt;/code&gt;&lt;/li&gt;
+	&lt;li&gt;&lt;code&gt;Eclipse-BundleShape: dir&lt;/code&gt;&lt;/li&gt;
+&lt;/ul&gt;
+&lt;/br&gt;
+* If it becomes necessary to retrieve the file and keep the bundle as a jar, the implementation of the file locating needs to be adjusted and support a bundle directory.
+&lt;/p&gt;
+      </documentation>
+   </annotation>
+
+   <element name="extension">
+      <annotation>
+         <appinfo>
+            <meta.element />
+         </appinfo>
+      </annotation>
+      <complexType>
+         <sequence>
+            <element ref="schemaAssociations" minOccurs="1" maxOccurs="unbounded"/>
+         </sequence>
+         <attribute name="point" type="string" use="required">
+            <annotation>
+               <documentation>
+                  The identifier of the target extension point
+               </documentation>
+            </annotation>
+         </attribute>
+         <attribute name="id" type="string">
+            <annotation>
+               <documentation>
+                  An optional identifier of the extension instance
+               </documentation>
+            </annotation>
+         </attribute>
+         <attribute name="name" type="string">
+            <annotation>
+               <documentation>
+                  An optional name of the extension instance
+               </documentation>
+               <appinfo>
+                  <meta.attribute translatable="true"/>
+               </appinfo>
+            </annotation>
+         </attribute>
+      </complexType>
+   </element>
+
+   <element name="schemaAssociations">
+      <annotation>
+         <documentation>
+            Sequence of all JSON schema associations
+         </documentation>
+      </annotation>
+      <complexType>
+         <sequence minOccurs="1" maxOccurs="unbounded">
+            <sequence minOccurs="1" maxOccurs="unbounded">
+               <choice>
+                  <element ref="bundleFilePath"/>
+                  <element ref="url"/>
+               </choice>
+            </sequence>
+            <sequence minOccurs="1" maxOccurs="unbounded">
+               <element ref="filePattern"/>
```

Comment:
Like everywhere else in Generic Editor and LSP4E, we should use content-types instead of direct file patterns. Content-types provide ability to match patterns and are reusable.